### PR TITLE
S28-1978 Add metric alerts

### PR DESCRIPTION
**Pull request by zc-hmcts**
### Change description ###
add metric-based alerts for postgres
add metric-based alerts for ingest and final storage accounts

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by jasonpaige — APPROVED** [review]

File: main.tf [hunk 1]
```diff
@@ -60,3 +60,28 @@ resource "azurerm_automation_account" "pre-aa" {
 
   tags = var.common_tags
 }
+
+data "azurerm_key_vault_secret" "slack_monitoring_address" {
+  name         = "slack-monitoring-address"
+  key_vault_id = data.azurerm_key_vault.keyvault.id
+}
+
+resource "azurerm_application_insights" "this" {
```
Comment (jasonpaige) [inline main.tf:69]: although it's not referenced else where, would it be better to name this `"ai"` or something?

File: main.tf [hunk 2]
```diff
@@ -60,3 +60,28 @@ resource "azurerm_automation_account" "pre-aa" {
...
+  count               = var.env == "prod" ? 1 : 0
+  name                = "pre-${var.env}-appinsights"
+  location            = data.azurerm_resource_group.rg.location
+  resource_group_name = data.azurerm_resource_group.rg.name
+  application_type    = "other"
+}
+
```
Comment (jasonpaige) [inline main.tf:76]: Can we add the instrumentation key to the KV too please? need it in the API app

```
resource "azurerm_key_vault_secret" "AZURE_APPINSIGHTS_KEY" {
  name         = "AppInsightsInstrumentationKey"
  value        = azurerm_application_insights.appinsights.instrumentation_key
  key_vault_id = module.key-vault.key_vault_id
}
```

**Review by jasonpaige — CHANGES_REQUESTED** [review]

File: main.tf (hunk 2 above)
Comment (zc-hmcts) [reply main.tf:76]: sure can, will raise this in a separate pr

File: main.tf (hunk 1 above)
Comment (zc-hmcts) [reply main.tf:69]: its a convention used in the terraform best practices guide https://www.terraform-best-practices.com/naming and so if there isn't a more descriptive name for the resource, "this" is preferred

Comment (jasonpaige) [reply main.tf:69]: TIL - thanks

**Pull request merged**